mux.go

```diff
@@ -224,8 +225,24 @@ func (tr treeRouter) ServeHTTPC(ctx context.Context, w http.ResponseWriter, r *h
 		routePath = r.URL.Path
 	}
 
+	// Check if method is supported by chi
+	method, ok := methodMap[r.Method]
+	if !ok {
+		// Return allowed methods, as required by RFC2616 for 405 Method not allowed
+		methods := make([]string, len(methodMap))
```

Comment:
thanks for fixing this panic.

I think we should move this procedure to its own function, like "writeMethodNotAllowed(w); return" .